dbms/src/Functions/FunctionsRound.h

```diff
@@ -1219,6 +1263,14 @@ struct TiDBRoundPrecisionInferer
         }
 
         PrecType new_prec = std::min(decimal_max_prec, int_prec + int_prec_increment + new_scale);
+        if unlikely (new_prec == 0)
+        {
+            // new_prec can be zero when the prec is eq to scale and frac is zero, for example:
+            // select truncate(0.22, 0) from t_col_decimal_2_2;
+            // TiDB will do fold constant, so this branch should reach here, but we still handle it.
```

Comment:
Can't understand why we still need to handle it when tidb already fold the constant. After constant folding, the truncate still exist?

Reply:
This unit test(https://github.com/pingcap/tiflash/pull/9842/files#diff-36ddb93c7399a0d4f862b2d92f01a11a84474e5c0ca2985f603c9167acc00cb8R327) will fail if we didn't handle